bin/run-docker

```diff
@@ -4,16 +4,40 @@ set -e
 
 uname_S=$(uname -s 2>/dev/null || echo not)
 
+
+
+
+
 ARDUINO_LOCAL_LIB_PATH="${ARDUINO_LOCAL_LIB_PATH:-${HOME}/Arduino}"
 
 if [ "${uname_S}" = "Darwin" ]; then
     ARDUINO_LOCAL_LIB_PATH="${ARDUINO_LOCAL_LIB_PATH:-${HOME}/Documents/Arduino}"
+    # This stops macos from copying resource forks into thigns like tar
+    export COPYFILE_DISABLE=true
 fi
 
 BOARD_HARDWARE_PATH="${BOARD_HARDWARE_PATH:-${ARDUINO_LOCAL_LIB_PATH}/hardware}"
 
-docker build -t kaleidoscope/docker etc
+
+XFER_DIR="$(pwd)/.docker_xfer"
+
+mkdir -p "${XFER_DIR}"
+
+tar  -cf "${XFER_DIR}/kaleidoscope.tar"  \
+	--exclude .docker_xfer	--exclude .git --exclude _build --exclude testing/googletest/build  \
+	.
+
+(cd "${BOARD_HARDWARE_PATH}/keyboardio" && tar  -cf "${XFER_DIR}/bundle.tar" \
+	--exclude .git --exclude avr/libraries/Kaleidoscope .)
+
+docker build -q -t kaleidoscope/docker etc
```

Comment:
I originally had `-q` in there, but building the docker image can take quite a bit of time too, so I dropped it. I agree that it's too verbose with it, but just staying completely silent until we actually run the image and have the entry point do status reports isn't great.

Reply:
*nod* a status message is fine. (really, I sort of wish we weren't calling docker builder every run, since it adds a second to runtime for me. But it's not worth effort)